sdk/core/core/inc/az_span.h

```diff
@@ -135,161 +158,297 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
     }, \
   }
 
+/**
+ * @brief AZ_SPAN_FROM_INITIALIZED_BUFFER returns an az_span expression over an initialized byte
+ * buffer. For example
+ *
+ * uint8_t buffer[] = { 1, 2, 3 };
+ * const az_span buf = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER(buffer); // Len=3, Cap=3
+ */
 #define AZ_SPAN_FROM_INITIALIZED_BUFFER(BYTE_BUFFER) \
   (az_span) AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER(BYTE_BUFFER)
 
 /******************************  SPAN MANIPULATION */
 
 /**
- * @brief Returns a sub span of the given span.
+ * @brief az_span_slice returns a new az_span which is a sub-span of the specified span.
  *
+ * @param[in] span The original az_span
+ * @param[in] low_index An index into the original az_span indicating where the returned az_span
+ * will start
+ * @param[in] high_index An index into the original az_span indicting where the returned az_span
+ * should stop. The byte at the high_index is NOT included in the returned az_span.
+ * @return An az_span (view) into a portion (from low_index to high_index - 1) of the original
+ * az_span.
  */
 AZ_NODISCARD az_span az_span_slice(az_span span, int32_t low_index, int32_t high_index);
```

Comment:
We modeled after Go' slices but also how would you get the count? Typically, you scan forward in the buffer for the next occurrence of some byte until you find its offset and then you slice from the start index to the found index (exclusive). If you make it a count, then the customer has to do the math which is more error prone. I think we addressed the more common use case in a better way.

I'm fine with start_index & end_index instead of low/high.